src/featurize.py

```diff
@@ -0,0 +1,125 @@
+import numpy as np
+import scipy as sp
+import sklearn
+import sklearn.preprocessing
+import copy
+
+def generate_feature_matrix(data, degree=2):
+
+    """
+    Function to generate a featurized data matrix from a data matrix where each
+    row is a scattering event. We use polynomial feature with arbitrary degree
+    as well as dot product of different ion/electron momenta.
+
+    Parameters
+    ------------
+    data : np.array [todo] How does this work if I pass in a pandas array
+        Array containing a row for each scattering event. We have 15 columns
+        (px, py, pz) for each of 5 scattering products: ion 1, ion 2, neutral,
+        electron 1, electron 2.
+    degree : int
+        Degree of the featurization desired. For $n$ features, a degree $d$
+        featurization will have $(d+n)C(d) = (d+n)C(n)$ features. With $n$
+        fixed, this scales as $d^n$. We will include a constant feature of one.
+
+    Returns
+    --------
+    array or dataframe
```

Comment:
Does this function currently return an array or dataframe? if so in what circumstances? It doesn't seam like both cases are tested.

Reply:
Only works with np.array. This has been clarified.